### PR TITLE
add : watch later functionality and UI modifications

### PR DESCRIPTION
**Pull request by ApurvaSawant11**
In this PR, I have added watch later functionality and made a few UI changes.

### Files to check:
- watch-later-services.js
- SaveModal.jsx
- modal.css

### Files to ignore:
- rest of the CSS files can be ignored
- minor modifications

### Please review and suggest

### CODE REVIEW

File: src/services/watch-later-services.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    dispatch({
+      type: "SET_WATCH_LATER",
+      payload: watchlater,
+    });
+  } catch (error) {
+    console.error("Error in addToWatchLater: ", error);
+  }
+};
+
+const removeFromWatchLater = async (dispatch, video, token) => {
```
Comment (foolhardy21) [inline src/services/watch-later-services.js]: as you only need `_id` for remove api, you can avoid passing the whole `video` object.

File: src/components/modal/SaveModal.jsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  };
+
+  return (
+    <div
+      onClick={() => setShowModal(!showModal)}
+      className={"flex-row-center modal-container"}
+    >
+      <div
+        onClick={(event) => {
+          event.stopPropagation();
```
Comment (foolhardy21) [inline src/components/modal/SaveModal.jsx:25]: good use of `stopPropagation`.

**Review by foolhardy21 — APPROVED** [review]
The code is clean and readable. left one suggestion. good work.

File: src/components/header/Header.jsx [hunk 3]
```diff
@@ -9,26 +9,16 @@ const Header = ({ showSidebar, setShowSidebar }) => {
...
-
   useEffect(() => {
     setShowSidebar(
-      location.pathname.includes("/watch")
-        ? "collapsed display-none"
-        : getSidebarClass(window.innerWidth)
+      location.pathname.includes("/watch/") || window.innerWidth < 760
+        ? "display-none"
+        : ""
     );
```
Comment (nayyyhaa) [inline src/components/header/Header.jsx:17]: suggestion: try to give dependency array in useEffect

**Review by nayyyhaa — APPROVED** [review]
Good is well written with proper ES6 standards, reusable components & well defined folder structure. UI looks awesome too! You can merge the PR 💯

File: src/components/header/Header.jsx (hunk 3 above)
Comment (ApurvaSawant11) [reply src/components/header/Header.jsx:17]: This useEffect already has a dependency array

File: src/services/watch-later-services.js (hunk 1 above)
Comment (ApurvaSawant11) [reply src/services/watch-later-services.js]: Yes! Thanks for pointing it out. I have made the changes

**Pull request merged**